src/components/Table/Table.tsx

```diff
@@ -193,19 +225,46 @@ function Table<T>({
                     dataTestIdPrefix={dataTestIdPrefix}
                   />
                 )}
-                {columns.map((item, index) => (
-                  <TableCell
-                    textAlign={columnsHasNumberArr && columnsHasNumberArr[index] ? 'right' : 'left'}
-                    component={'th'}
-                    key={`${item}`}
-                    sticky={fixedHeader}
-                    padded={padded}
-                    width={columnsWithWidth[index] ? `${columnsWithWidth[index]}%` : 'initial'}
-                    dataTestIdPrefix={dataTestIdPrefix}
-                  >
-                    {item}
-                  </TableCell>
-                ))}
+                {/* TODO: Remove this when the open TS issue, regarding arrays with multiple types, is fixed */}
+                {/* eslint-disable-next-line @typescript-eslint/ban-ts-comment */}
+                {/* @ts-ignore */}
+                {columns.map((item: string | ExtendedColumn, index: number) => {
+                  return (
+                    <TableCell
+                      textAlign={
+                        columnsHasNumberArr && columnsHasNumberArr[index] ? 'right' : 'left'
+                      }
+                      component={'th'}
+                      key={`${typeof item === 'string' ? item : item.content}`}
```

Comment:
`typeof item === 'string'` as this is used many times create a const maybe called `itemIsTypeString` and use it accordingly 
`itemIsTypeString`
`!itemIsTypeString`